File: src/main/javabak/multiblock/MultiblockPart.java
package redcrafter07.processed.multiblock;

import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.Blocks;

import java.util.HashMap;
import java.util.Map;

public enum MultiblockPart {
    Ignored,
    Empty,
    Casing,
    Controller;

    public static Map<Block, MultiblockPart> map(Block wall, Block controller) {
        final Map<Block, MultiblockPart> map = new HashMap<>();

        map.put(wall, MultiblockPart.Casing);
        map.put(controller, MultiblockPart.Controller);
        map.put(Blocks.AIR, MultiblockPart.Empty);

        return map;
    }
}

File: src/main/kotlin/redcrafter07/processed/multiblock/MultiblockPart.kt
package redcrafter07.processed.multiblock

import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.Blocks

enum class MultiblockPart {
    Ignored, Empty, Casing, Controller;

    companion object {
        fun makeBlockMap(wall: Block, controller: Block): Map<Block, MultiblockPart> = mapOf(
            Pair(Blocks.AIR, Empty),
            Pair(wall, Casing),
            Pair(controller, Controller),
        )
    }
}